include/IStorageImage.h

```diff
@@ -26,7 +26,7 @@ class IStorageImage : public virtual IVulkanPrebaked
         {
```

Comment:
This file/class shouldn't exist, there's no such thing as a "Storage Image" object in Vulkan, its just a binding in the descriptor set.
https://www.khronos.org/registry/vulkan/specs/1.1-extensions/man/html/VkDescriptorImageInfo.html

I know you need to create `ImageView`  OpenGL objects, but like VAOs they are non shareable between contexts, so need a driver-wide cache for them :(  with the key being `pair<IRenderableVirtualTexture*,GLuint/*Texture ID*/>` 

There is however a `ImageView` (same as Texture View in OpenGL) but thats my responsibility to implement.

Reply:
(it's not my file, all mine lives in include/irr/...)
However.. there is VkImage https://vulkan.lunarg.com/doc/view/1.0.33.0/linux/vkspec.chunked/ch11s03.html and we need some class describing such binding (texture object + something `VkImageCreateInfo`-like) in order to bind "storage images" (GLSL's `image2D` etc). Otherwise we can't know miplevel and layer to bind.

Reply:
Ok so nuke this file, its probably one of my stubs.

Also you don't need a VkImageCreateInfo, just `vkDescriptorImageInfo`
http://vulkan-spec-chunked.ahcox.com/ch13s02.html

Reply:
You can ignore ImageLayout for now, but take a loot into this:
- If descriptorType is VK_DESCRIPTOR_TYPE_COMBINED_IMAGE_SAMPLER, VK_DESCRIPTOR_TYPE_SAMPLED_IMAGE, VK_DESCRIPTOR_TYPE_STORAGE_IMAGE or VK_DESCRIPTOR_TYPE_INPUT_ATTACHMENT, the imageView and imageLayout members of any given element of pImageInfo must be a valid VkImageView and VkImageLayout, respectively 
- If descriptorType is VK_DESCRIPTOR_TYPE_STORAGE_IMAGE or VK_DESCRIPTOR_TYPE_INPUT_ATTACHMENT, the imageView member of any given element of pImageInfo must have been created with the identity swizzle

Basically in Vulkan, you cannot use a texture (VkImage) directly, only via ImageViews (OpenGL Texture Views). This allows you to use ARB_multibind for storage images because the mip level and layer is always 0 ;)

Create a stub `asset::ImageView`,`asset::ICPUImageView` and `video::IGPUImageView` but only as a stub... I'll sort it out later in  #148 and https://github.com/buildaworldnet/IrrlichtBAW/projects/15